### PR TITLE
Started a module, using CIM instead of WMI, remote support, credentials

### PR DESCRIPTION
**Pull request by KevinMarquette**
I started building out a module. 
- I took the WMI settings approach and reworked it to use CIM session. 
- Added auto discovery of SSRS version. 
- Switched from username and password strings to using credentials 
- Added remote support with `-ComputerName` parameter
- Added pipeline support for `-ComputerName` parameter for easier batch targeting groups of machines
- Created new cmdlets that exposes all CIM Methods. Still need to review if some are just clutter and should be trimmed out later.

I know this is a large pull request but it should be all new files in a module folder. The existing standalone scripts are untouched. If we want to continue down the module route, those scripts would be easy to work into it. This would also solve the "install" scenario for the users.

### CODE REVIEW

**Comment by parthsha** [comment]
Can you please add instructions on how to download the additional tools (PSake, BuildHelpers, and Pester) by updating the Readme.MD in the Scripts folder?

**Comment by KevinMarquette** [comment]
Yes I will. They are all available via the `Find-Module` and `Install-Module` command from PSGet but I will add that documentation.

File: PSSSRS/public-functions/Get-RSConfigurationSetting.ps1 [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Licensed under the MIT License (MIT)
+#>
+
+function Get-RSConfigurationSetting
+{
+<#
+.SYNOPSIS
+
+.EXAMPLE
+New-RSConfigurationSession -ComputerName server
```
Comment (jtarquino) [inline PSSSRS/public-functions/Get-RSConfigurationSetting.ps1]: the example is for +New-RSConfigurationSession, should be updated to Get-RSConfigurationSetting

**Comment by jtarquino** [comment]
Hi Kevin, thanks for this great contribution, to be honest we are learning how to do this with you and the community and we were really surprised by the passion and effort you put on it, we really appreciate it although to be honest we were expecting incremental improvements and because it was a big surprise contribution it took us a while to make some time aside of our current efforts (check out https://youtu.be/gkOG9HxRpZs for more details).

I was in the process of merging the pull request and found out the readme.md is still pending, from recent conversations with users of the existing scripts the thing that they have found more challenging is how setup the scripts to start using it, so the get started is key to allow the users to use it

File: PSSSRS/public-functions/Add-RSReservedURL.ps1 [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+Licensed under the MIT License (MIT)
+#>
+
+function Add-RSReservedURL
+{
+<#
+.SYNOPSIS
+Adds a URL reservation for a given application.
+.EXAMPLE
+Set-RSReservedURL -Application ReportManager -UrlString 'http://+:80'
```
Comment (jtarquino) [inline PSSSRS/public-functions/Add-RSReservedURL.ps1]: example should be Add-RSReservedURL

File: PSSSRS/public-functions/Add-RSSSLCertificateBinding.ps1 [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+{
+<#
+.SYNOPSIS
+Create SSL Certificate Bindings
+.EXAMPLE
+Add-RSSSLCertificateBinding -Application ReportManager
+.EXAMPLE
+ 
+.NOTES
+ CreateSSLCertificateBinding(
```
Comment (jtarquino) [inline PSSSRS/public-functions/Add-RSSSLCertificateBinding.ps1]: notes doesn't match function name

File: PSSSRS/public-functions/Get-RSAdminSiteUrl.ps1 [hunk 4]
```diff
@@ -0,0 +1,63 @@
+<#
+Copyright (c) 2016 Microsoft Corporation. All Rights Reserved.
+Licensed under the MIT License (MIT)
+#>
+
+function Get-RSAdminSiteUrl
+{
+<#
+.SYNOPSIS
+Get Admin Site Url
```
Comment (jtarquino) [inline PSSSRS/public-functions/Get-RSAdminSiteUrl.ps1]: probably a small comment that this method only returns a site when SSRS is SharePoint integrated

File: PSSSRS/public-functions/New-RSDatabaseCreationScript.ps1 [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+Licensed under the MIT License (MIT)
+#>
+
+function New-RSDatabaseCreationScript
+{
+<#
+.SYNOPSIS
+Generates a SQL Script that can be used to create a report server database.
+.EXAMPLE
+ew-RSDatabaseCreationScript
```
Comment (jtarquino) [inline PSSSRS/public-functions/New-RSDatabaseCreationScript.ps1]: nit: missing a N

**Comment by KevinMarquette** [comment]
No problem. I know your guys have a lot on your plate especially with
Ignite. I didn't make it out to the conference but I am still watching
sessions from that. Sorry for the overwhelming number of changes. I was
looking for a weekend project and the timing of this was perfect for that.
I just have not looped back to this until now.

With that said, I see you moved the scripts to a new repo and have already
integrated a lot of the ideas I was putting forward. I think what I should
do now is close that existing pull request and clone the new project. I
think the CIM based commands and my approach to the cmdlet parameters still
has value to the community. I'll work them into the new project and open a
new pull request. My plan after that is to build a DSC resource for a few
of them.

**Comment by jtarquino** [comment]
Yes!!!, the new repo is for simplified installation , the CIM commands to access WMI looks pretty interesting and the ability to execute remote WMI sounds like a great next step, the only thing I will suggest to think about is that when I tried your PR it ask for credentials for every function (everybody loves to be asked for credentials every single time as much as everybody loves 2FA or complex passwords with at least 25 characters long ;))

**Comment by KevinMarquette** [comment]
Great feedback. I was crossing domains to a testlab in my testing so I
needed to specify the creds. I'll figure out why it was requiring it. I
truly intended for it to be optional just like the base WMI commands, even
for remote systems.

Publishing to the PS gallery is another option for ease of installation.

**Comment by jtarquino** [comment]
Yes we plan to do it also in the gallery, still figuring out the best way to keep that one in sync automatically with the repo

**Comment by KevinMarquette** [comment]
Here is one approach that I am aware of:
http://ramblingcookiemonster.github.io/PSDeploy-Inception/

It's on my todo list too for my own modules.

**Comment by jtarquino** [comment]
Hi Kevin closing the PR from our last conversation.
BTW the new repo also publish to the gallery so the new cmdlets can be installed using install-module ReportingServicesTools

**Pull request closed without merging**